crates/anvil/src/eth/api.rs

```diff
@@ -1312,6 +1318,61 @@ impl EthApi {
         .map(U256::from)
     }
 
+    /// Handler for RPC call: `anvil_getBlobByHash`
+    pub async fn anvil_get_blob_by_versioned_hash(
+        &self,
+        hash: B256,
+    ) -> Result<Option<alloy_consensus::TxEip4844WithSidecar>> {
+        node_info!("anvil_getBlobByHash");
+        let latest_block = self.backend.best_number();
+        for i in 0..=latest_block {
+            if let Some(block) = self.backend.get_full_block(i) {
+                for tx in block.into_transactions_iter() {
+                    let typed_tx = TypedTransaction::try_from(tx).unwrap();
+
+                    if let TypedTransaction::EIP4844(signed) = typed_tx
+                        && let Ok(sidecar_tx) = signed.tx().clone().try_into_4844_with_sidecar()
+                    {
+                        for item in sidecar_tx.sidecar.clone() {
+                            let versioned_hash = B256::from(item.to_kzg_versioned_hash());
+
+                            if versioned_hash == hash {
+                                return Ok(Some(sidecar_tx));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    /// Handler for RPC call: `anvil_getBlobByTransactionHash`
+    pub async fn anvil_get_blob_by_tx_hash(
+        &self,
+        hash: B256,
+    ) -> Result<Option<alloy_consensus::TxEip4844WithSidecar>> {
```

Comment:
for this we should be able to reuse this logic:

https://github.com/foundry-rs/foundry/blob/d592b3e0f142d694c3be539702704a4a73238773/crates/anvil/src/eth/backend/mem/mod.rs#L2903-L2909

we should add a helper fn for this to Backend as well